armi/reactor/reactors.py

```diff
@@ -2287,8 +2276,6 @@ def processLoading(self, cs, dbLoad: bool = False):
             if not cs["detailedAxialExpansion"]:
                 # prepare core for mesh snapping during axial expansion
                 for a in self.getAssemblies(includeAll=True):
-                    if a.hasFlags(nonUniformAssems, exact=True):
-                        continue
```

Comment:
Needed to make the `axialSnapList` for the non-uniform assemblies to snap them to the same mesh as the core when thermal expansion is on (i.e., `inputHeightsConsideredHot: false`). This is another piece of the PR that I am planning to do more testing on and see if it can be reverted or somehow handled differently.